task.dart

```diff
@@ -0,0 +1,116 @@
+import 'dart:io';
+
+class Task {
+  int id;
+  String? title;
+  String? description;
+  bool isCompleted;
+
+  Task(this.id, this.title, this.description, this.isCompleted);
+}
+
+class TaskManager {
+  List<Task> tasks = [];
+  int id = 1;
+
+  //Add tasks
+  void addTask(String? title, String? description, bool isCompleted) {
+    tasks.add(Task(id, title, description,
+        isCompleted)); //takes constructor/object as a input
+
+    id++;
+
+    print("Task successfully added.");
+    print("\n****************************************");
+  }
+
+  //Display tasks
+  void display() {
+    print("\n**************Display tasks***************");
+    if (tasks.isNotEmpty) {
+      for (int i = 0; i < tasks.length; i++) {
+        Task task = tasks[i];
+        print(
+            "Task ID: ${task.id}\nTask name: ${task.title}\nTask Description: ${task.description}\nTask completed: ${task.isCompleted}");
+        print("\n*****************************************");
+      }
+    } else {
+      print("No tasks present.");
+      print("\n*******************************************");
+    }
+  }
+
+  //Delete tasks
+  void delete(taskid) {
+    print("\n**************Delete Task**************");
+    if (tasks.isNotEmpty) {
+      for (int i = 0; i < tasks.length; i++) {
+        Task task = tasks[i];
+        if (task.id == taskid) {
+          tasks.removeAt(i);
+          print("Task Removed Successfully.");
+          print("\n***************************************");
+          break;
+        } else if (i == tasks.length - 1) {
+          print("ID not present.");
+          print("\n***************************************");
+        }
+      }
```

Comment:
instead of looping through all the tasks you can use the .removeWhere
```suggestion
      tasks.removeWhere((task) => task.id == taskId)
```